src/altair/helpers.rs

```diff
@@ -144,11 +204,47 @@ pub fn get_unslashed_participating_indices<
         MAX_VALIDATORS_PER_COMMITTEE,
         SYNC_COMMITTEE_SIZE,
     >,
-    _flag_index: usize,
-    _epoch: Epoch,
-    _context: &Context,
+    flag_index: usize,
+    epoch: Epoch,
+    context: &Context,
 ) -> Result<HashSet<ValidatorIndex>> {
-    Ok(Default::default())
+    // Return the set of validator indices that are both active and unslashed for the given ``flag_index`` and ``epoch``
+    let previous_epoch = get_previous_epoch(state, context);
+    let current_epoch = get_current_epoch(state, context);
+    let is_current = epoch == current_epoch;
+    if previous_epoch != epoch && current_epoch != epoch {
+        return Err(Error::InvalidEpoch {
+            requested: epoch,
+            previous: previous_epoch,
+            current: current_epoch,
+        });
+    }
+
+    let active_validator_indices = get_active_validator_indices(state, epoch);
+    let participating_indices = active_validator_indices
+        .iter()
+        .filter_map(|&i| {
```

Comment:
```suggestion
        .filter(|&i| {
```

and then simplify the predicate body